docs/quality_assurance.md

```diff
@@ -0,0 +1,124 @@
+# Interchain Security: Quality Assurance
+
+This document contains the overview of the quality assurance process necessary for the release of Interchain Security v1. 
+
+The verification of Interchain Security is done using the following approaches:
+- code review
+- automatic software engineering tools (e.g., SonarCloud, gosec)
+- unit testing
+- differential testing
+- incentivized testnet
+- audit (with the Apalache team from Informal Systems)
+
+The verification of Interchain Security is split across the following concerns:
+- Correct software engineering practices (e.g., error handling, serialization, deserialization).
+- The correct integration with IBC (i.e., [ibc-go](https://github.com/cosmos/ibc-go/tree/v3.0.0)).
+- The correct integration with Cosmos SDK (i.e., [cosmos-sdk](https://github.com/cosmos/cosmos-sdk/tree/v0.45.6)).
+- The correctness of the provider chain, i.e., the provider CCV module does not break the liveness or the safety of the provider chain. 
+- The correctness of the Interchain Security protocol, i.e., the protocol follows the [specification](https://github.com/cosmos/ibc/blob/master/spec/appics-028-cross-chain-validation/README.md).
+- The correctness of the consumer chain, i.e., both liveness or safety hold.
+
+> TODO decide what should be covered by the audit and the priority
+
+## Software Engineering
```

Comment:
@danwt @sainoe Is there something else that we should check for in this section?

Reply:
LGTM